proxy.go

```diff
@@ -57,14 +79,17 @@ func ProxyHandler(w http.ResponseWriter, req *http.Request) {
 		authenticated, err := http.NewRequest("GET", req.URL.String(), req.Body)
 		authenticated.Header.Set("On-Behalf-Of", user)
 		var solutionMsg string
+		var client *http.Client
 		// Retry the request
 		if len(cookies[user]) > 0 { // Use existing cookie
 			authenticated.AddCookie(cookies[user][req.Host][0])
 			// Create the client
-			client = NewClient(insecureSkipVerify)
+			// client = NewClient(insecureSkipVerify)
+			client = p.HttpClient
 			solutionMsg = "Retrying with cookies"
 		} else { // Using WebIDTLS client
-			client = NewAgentClient(agentCert)
+			// client = NewAgentClient(agentCert)
+			client = p.HttpAgentClient
 			solutionMsg = "Retrying with WebID-TLS"
 		}
 		r, err = client.Do(authenticated)
```

Comment:
Just noting that I spotted another explicit HTTP 500 below... we can probably come up with more semantic errors.